generators/server/index.js

```diff
@@ -259,6 +264,11 @@ module.exports = JhipsterServerGenerator.extend({
                     this.serviceDiscoveryType = false;
                 }
 
+                // If the service discovery is not defined, it is disabled by default
+                if (this.apiFirst === undefined) {
```

Comment:
I copy/pasted the behavior from serviceDiscoveryType (even the comment, I realize, ...) but you're probably right.